code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,401 @@
+	var/mob_content_size = mob_to_insert.mob_size
+	if(max_tub_capacity < ((mob_content_size * 2) + current_tub_capacity))
+		to_chat(user, "<span class='warning'>You try to insert [mob_to_insert] into [src] but the tub is too full for [mob_to_insert.p_them()]!</span>")
+		return FALSE
+	to_chat(mob_to_insert, "<span class='userdanger'>[user] starts shoving you into [src]!</span>")
+	visible_message("<span class='warning'>[user] starts shoving [mob_to_insert] into [src]!</span>")
+	if(!do_after_once(user, (3 SECONDS * mob_content_size), target = src, attempt_cancel_message = "You stop inserting [mob_to_insert] into [src]."))
+		return FALSE
+	mob_to_insert.forceMove(src)
+	RegisterSignal(mob_to_insert, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	LAZYADD(inserted_mobs, mob_to_insert)
+	calculate_tub_capacity()
+	update_washing_state()
+	add_attack_logs(user, mob_to_insert, "Shoved into washing machine.")
+	to_chat(mob_to_insert, "<span class='userdanger'>[user] shoves you into [src]. Oh shit!</span>")
+	to_chat(user, "<span class='danger'>You shove [mob_to_insert] into [src].</span>")
+	return TRUE
+
+/// Handles inserting obj/items into the washing machines, checks machines capacity, does a do_after, and then applys appropriate signals and updates machines state
+/obj/machinery/washing_machine/proc/insert_item_into_tub(mob/user, obj/item/item_to_insert)
+	var/item_content_size = item_to_insert.w_class
+	if(max_tub_capacity < (item_content_size + current_tub_capacity))
+		to_chat(user, "<span class='warning'>You try to insert [item_to_insert] into [src] but the tub is too full for [item_to_insert.p_them()]!</span>")
+		return FALSE
+	if(!do_after_once(user, (1 SECONDS * (item_to_insert.w_class / 3)), target = src, attempt_cancel_message = "You stop inserting [item_to_insert] into [src]."))
+		return FALSE
+	if(!user.drop_item())
+		to_chat(user, "<span class='warning'>[item_to_insert] is stuck to your hand!</span>")
+		return FALSE
+	item_to_insert.forceMove(src)
+	RegisterSignal(item_to_insert, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	LAZYADD(inserted_items, item_to_insert)
+	calculate_tub_capacity()
+	update_washing_state()
+	to_chat(user, "<span class='notice'>You insert [item_to_insert] into [src].</span>")
+	return TRUE
+
+/*
+	* # check_tub_contents()
+	*
+	* Managing tons of contents inside the object is messy business, you never know when something is deleted, teleported, or moved into your obj
+	* without your machine knowing, this clears bad references for GC/tracking purposes and adds references when something is added randomly to
+	* our contents. This ensures we don't nullspace shit (such as a mob dropping a wrench or something) and spit it all back out afterwards even
+	* if we didn't add it our selves!
+	*
+	* This proc is called via COMSIG_PARENT_QDELETING for objects we put inside through the insert proc
+*/
+/obj/machinery/washing_machine/proc/check_tub_contents()
+	if(LAZYLEN(inserted_items))
```

Comment:
this whole proc chain could probably use some stack_traces

Reply:
I mean, not really. The whole point of this proc is to detect when the contents of the washing machine don't match its tracked lists. This can happen for a number of reasons that aren't bugs and is 100% expected behavior.